mutagen/id3/_file.py

```diff
@@ -149,13 +153,18 @@ class XMYF(Frame): ...
         try:
             self._header = ID3Header(fileobj)
         except (ID3NoHeaderError, ID3UnsupportedVersionError):
-            frames, offset = find_id3v1(fileobj)
+            if not load_v1:
+                raise
+
+            frames, offset = find_id3v1(fileobj, v2_version)
             if frames is None:
                 raise
 
             self.version = ID3Header._V11
-            for v in frames.values():
-                self.add(v)
+            for k, v in frames.items():
```

Comment:
Some issues with this:

* known_frames can map existing frame keys to different frame classes, so while the old code ignored known_frames this ignores the value part of it and always uses the builtin frame types anyway. I guess it's still an improvement though.
* besides that, frames is a HashKey->Frame mapping while known_frames is a FrameID->Frame mapping, so this will throw away all COMM frames from ParseID3v1 for example.
* If you keep the known_frames change can you add a small test?